artifactory/services/release.go

```diff
@@ -0,0 +1,84 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	ioutils "github.com/jfrog/jfrog-client-go/utils/io"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"strings"
+)
+
+const (
+	conflictErrorMessage  = "Bundle already exists"
+	importRestApiEndpoint = "api/release/import"
+	// TODO change this when the bug is fixed
+	// https://jfrog-int.atlassian.net/browse/JR-8542
+	tempImportRestApiEndpoint = "ui/api/v1/ui/release/import"
+	octetStream               = "application/octet-stream"
+)
+
+type ReleaseService struct {
+	client     *jfroghttpclient.JfrogHttpClient
+	ArtDetails auth.ServiceDetails
+	Progress   ioutils.ProgressMgr
+}
+
+type ErrorResponseWithMessage struct {
+	Errors []ErrorDetail `json:"errors"`
+}
+
+type ErrorDetail struct {
+	Status  int    `json:"status"`
+	Message string `json:"message"`
+}
+
+func NewReleaseService(artDetails auth.ServiceDetails, client *jfroghttpclient.JfrogHttpClient) *ReleaseService {
+	return &ReleaseService{client: client, ArtDetails: artDetails}
+}
+
+func (rs *ReleaseService) GetJfrogHttpClient() *jfroghttpclient.JfrogHttpClient {
+	return rs.client
+}
+
+func (rs *ReleaseService) ImportReleaseBundle(filePath string) (err error) {
+	// Load desired file
+	content, err := fileutils.ReadFile(filePath)
```

Comment:
Just to make sure - did you happen to check if the file name of the archive can be changed or is expected to be left unchanged?

Reply:
When you import a bundle you can't change it's name.
And even if you do, if the metadata inside the bundle doesn't match it fails to import. 👍

Reply:
I've added a unit test for this case.